### PR TITLE
add WebSocketClient

### PR DESCRIPTION
**Pull request by DmytroOrlov**

### CODE REVIEW

File: client/src/main/scala/client/WebSocketClient.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        def onComplete() = {
+          closeConnection()
+          subscriber.onComplete()
+        }
+      })
+  }
+}
+
+object WebSocketClient {
+  def apply(url: String): WebSocketClient = {
```
Comment (novakov-alexey) [inline client/src/main/scala/client/WebSocketClient.scala]: url: String is used here? If not, then remove it

File: client/src/main/scala/client/WebSocketClient.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Success, Try}
+
+class WebSocketClient(url: String) extends Observable[String] with (String => Unit) {
+  private val sendOnOpen = Promise[String => Unit]()
+
+  def apply(message: String): Unit = {
```
Comment (novakov-alexey) [inline client/src/main/scala/client/WebSocketClient.scala]: Does Scala treat apply in class somehow differently than apply function on object ?

File: client/src/main/scala/client/WebSocketClient.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import org.scalajs.dom
+import org.scalajs.dom.raw.MessageEvent
+import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
+
+import scala.concurrent.Promise
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Success, Try}
+
+class WebSocketClient(url: String) extends Observable[String] with (String => Unit) {
```
Comment (novakov-alexey) [inline client/src/main/scala/client/WebSocketClient.scala]: with (String => Unit) - is it anonymous trait? never seen it before in Scala :-)

File: client/src/main/scala/client/WebSocketClient.scala (hunk 1 above)
Comment (DmytroOrlov) [inline client/src/main/scala/client/WebSocketClient.scala]: fixed, thanks!

File: client/src/main/scala/client/WebSocketClient.scala (hunk 3 above)
Comment (DmytroOrlov) [inline client/src/main/scala/client/WebSocketClient.scala]: you can extend by `with Function0[String]` if you want too, but `(String => Unit)` is more scala-way

File: client/src/main/scala/client/WebSocketClient.scala (hunk 2 above)
Comment (novakov-alexey) [inline client/src/main/scala/client/WebSocketClient.scala]: ок

**Pull request merged**